### DISCUSSION TITLE
Thoughts about release note URLs

### DISCUSSION THREAD

**Discussion by thomas-mc-work** [mag37/dockcheck#106] [General]
I've just tested the latest release regarding the release notes feature. For me it didn't work as expected. My email body looked like this:

    lego-web-1  ->  url missing
    nextcloud-cron-1  ->  url missing
    nextcloud-db-1  ->  url missing
    www-web-1  ->  url missing

That wasn't very satisfying :-) Wouldn't it be better to match the release note URL by the image URL? Container names can vary, but the image source URL is unique.

Another thing I've noticed is that the release notes URLs aren't printed when running the check via the console. Is there a reason not to do that?

Despite that I think this could become a great and very useful feature! Thank you.

**Comment by mag37** [comment]
Thank you for testing and good points.

The `urls.list` was created as a template/example first but then it seemed like a good idea to just continue to append to it.

I agree, image names might have been more "standard" even though they also vary - many different sources of the same service. So they'd need one entry for each source. Could still work though, so good point.

And another reason is that the list of containers with updates are just container names currently, not referencing image so it was easier to just iterate through that.
Could still work if I rewrote it to add all names + images to an array/table of some sort. Though it's pretty cumbersome already to do the lookups/correlation.

Lastly - they're not printed in the main output because I considered it being tied to the notifications. But you're right, maybe there should be an option for it!  
I'll take it into consideration and add it as future work.  

Currently doing a overhaul of it all.

**Edit**: And about the `urls.list` - it was meant to be an example for every user to modify locally to fit their environment.

**Comment by mag37** [comment]
Please look at the [notify-rewrite branch](https://github.com/mag37/dockcheck/discussions/101) discussion for further points, changes and questions.

**Comment by thomas-mc-work** [comment]
> Lastly - they're not printed in the main output because I considered it being tied to the notifications. But you're right, maybe there should be an option for it!

Maybe the output could be unified?

**Comment by mag37** [comment]
> Maybe the output could be unified?

Yes it could, but I know some people have used the output for custom things - so I rather not clutter that.
Unifying is still a good option - as only people who add the `urls.list` to the script directory will trigger it. And then maybe have a flag to not print the urls to console for those who dont want it (keeping an extra list or just printing column1 isnt much work).

Good points and will consider for future changes! Thank you.
If you dont mind I'll transfer this to a discussion instead of an issue?

**Comment by thomas-mc-work** [comment]
> Unifying is still a good option - as only people who add the urls.list to the script directory will trigger it.

I've just now realized that this is a reqiurement in order to make it work 😅 So yeah, that condition would be good. Either that, or you reference the source file on github and fetch it every time the script is invoked. But those aspects don't touch the issue with on what to use as the lookup key.

**Comment by mag37** [comment]
Sorry I've not replied - I wanted to push the notification rewrite first and then see if I can work on this - just havnt found the time.
I'll try to get some testing done in the future.

**Comment by yoyoma2** [comment]
Added a line to urls.list to fix a "url missing". It's related to vaultwarden so put it right after vaultwarden-server. Since this topic is under discussion not sure what the approved procedure is. A PR seems like overkill to add one url.

diff --git a/notify_templates/urls.list b/notify_templates/urls.list
index 4195632..dbe5a09 100644
--- a/notify_templates/urls.list
+++ b/notify_templates/urls.list
@@ -6,6 +6,7 @@ apprise-api https://github.com/linuxserver/docker-apprise-api/releases
 homer https://github.com/bastienwirtz/homer/releases
 nginx https://github.com/docker-library/official-images/blob/master/library/nginx
 vaultwarden-server https://github.com/dani-garcia/vaultwarden/releases
+bruceforce-vaultwarden-backup https://github.com/Bruceforce/vaultwarden-backup/blob/main/CHANGELOG.md
 actual_server https://actualbudget.org/blog
 gotify https://github.com/gotify/server/releases
 traefik https://github.com/traefik/traefik/releases

**Reply by mag37** [reply]
You're very welcome to do PRs no matter size or importance 😄 
But I'll just hot-add this to main if you wish, no worries. Thank you!

**Reply by yoyoma2** [reply]
Hot add is perfect.
